app/views/core/LoadingProgress.vue

```diff
@@ -56,6 +56,16 @@
           const statuses = this.statuses.map((status) => {
             if (typeof status === 'boolean') {
               return status ? 0 : 100
+            } else if (typeof status === 'object' && status.length > 0) {
+              const reducedArray = status.reduce((total, toAdd) => {
```

Comment:
Is there a possibility that the first element is a boolean and is thus set as the initial value? Should `0` be explicitly set as the initial total?

Reply:
Actually I am a little confused. We are calling `reduce` on `status` and are then checking if `status` is a `boolean` within the reduce. Is there a type here?

Reply:
For your first question: That's a great point! It would certainly not be correct in that case, because it would be coerced into a `1` - I will update this 👍 Great catch.

Second question: Yeah this is a bit weird. We are basically trying to say that a bool represents `100` or `0`. So for the things that are loading that only have a concept of yes/no as a loaded state, we represent it as numbers. 

This gets a bit more complex when having both numbers AND booleans. The way this is done here is applying the same logic for turning a boolean into a number in the reduce. 

Do you think it would be clearer if a function would be used for the conversion, @AndrewJakubowicz ? Or maybe just comments explaining what we are doing?

Reply:
I like the improvements done. Left another comment.